### PR TITLE
add the final solutions for ex 3 and ex 4

### PR DESCRIPTION
**Pull request by lirany1**
Hi
@eyalstoler

my final solutions to  homework1 

thanks for your help
Liran

### CODE REVIEW

File: session1/solutions/assignment3/Rakefile [hunk 1]
```diff
@@ -1 +1,20 @@
 # assignment 3 solution
+
+require 'fileutils'
+namespace :rename_yml do
```
Comment (eyalstoler) [inline session1/solutions/assignment3/Rakefile]: This is wrong. namespace was supposed to be :files and the actual task :rename_yml

File: session1/solutions/assignment3/Rakefile [hunk 2]
```diff
@@ -1 +1,20 @@
...
+
+  desc "Rename yaml format to yml format"
+ task :files, [:path] do 
+
+patt=ENV["path"]
```
Comment (eyalstoler) [inline session1/solutions/assignment3/Rakefile]: What does the parameter `patt` mean?

File: session1/solutions/assignment3/Rakefile [hunk 3]
```diff
@@ -1 +1,20 @@
 # assignment 3 solution
+
```
Comment (eyalstoler) [inline session1/solutions/assignment3/Rakefile:2]: Running `rake rename_yml:files[config.yaml]` provided the following error:

```
The files in your folder are:
rake aborted!
NoMethodError: undefined method `+' for nil:NilClass
/Users/eyals/dev/coding-assignments/session1/Rakefile:9:in `block (2 levels) in <top (required)>'
Tasks: TOP => rename_yml:files
(See full trace by running task with --trace)
```

Comment (eyalstoler) [reply session1/solutions/assignment3/Rakefile:2]: please check this

File: session1/solutions/assignment4/Rakefile [hunk 4]
```diff
@@ -1 +1,42 @@
...
+  dexclude = ENV["EXCLUDE"]
+
+    # Find all of the files 
+    Dir.chdir mpath
+    files = FileList["*"]
+    puts "The full file list is:"
+    files.each{|file| 
+      puts  file }
+    
+   #files=files.map{|v|%Q{#{v}}}.join(' ')
```
Comment (eyalstoler) [inline session1/solutions/assignment4/Rakefile]: Remove comment

File: session1/solutions/assignment4/Rakefile [hunk 5]
```diff
@@ -1 +1,42 @@
...
+ #namespace :gziped_files do
+  
+  desc "Archive files to gzip to a specified path"
+  task :archive , [:M_PATH, :EXCLUDE]  do
+
+  mpath = ENV["M_PATH"]
+  dexclude = ENV["EXCLUDE"]
+
+    # Find all of the files 
+    Dir.chdir mpath
```
Comment (eyalstoler) [inline session1/solutions/assignment4/Rakefile]: There is no need to change dir's - any ruby/rake command can get either relative or full paths

File: session1/solutions/assignment4/Rakefile [hunk 6]
```diff
@@ -1 +1,42 @@
...
+    exlist = dexclude.split(",")
+    exlist.each {|ex|
+      files = files.exclude("*#{ex}")}
+    
+    puts"The files that will be compress are: #{files} ""\n"
+
+
+    files = files.exclude("build")
+
+    #pathNew=DPATH
```
Comment (eyalstoler) [inline session1/solutions/assignment4/Rakefile]: Remove all comments. they causes mess in the code

File: session1/solutions/assignment4/Rakefile [hunk 7]
```diff
@@ -1 +1,42 @@
 # assignment 4 solution
+ #namespace :gziped_files do
```
Comment (eyalstoler) [inline session1/solutions/assignment4/Rakefile]: Given this input: `rake archive[/tmp/folder,.tmp]`, I received the following error:

rake aborted!
TypeError: no implicit conversion of nil into String
/Users/eyals/dev/coding-assignments/session1/Rakefile:12:in `chdir'
/Users/eyals/dev/coding-assignments/session1/Rakefile:12:in `block in <top (required)>'
Tasks: TOP => archive
(See full trace by running task with --trace)

File: session1/solutions/assignment4/Rakefile [hunk 8]
```diff
@@ -1 +1,42 @@
...
+  task :archive , [:M_PATH, :EXCLUDE]  do
+
+  mpath = ENV["M_PATH"]
+  dexclude = ENV["EXCLUDE"]
+
+    # Find all of the files 
+    Dir.chdir mpath
+    files = FileList["*"]
+    puts "The full file list is:"
+    files.each{|file|
```
Comment (eyalstoler) [inline session1/solutions/assignment4/Rakefile]: What are you printing this for?

**Review by eyalstoler — COMMENTED** [review]
@lirany1 I was unable to properly run both assignments. Please check why and fix. Also please review additional code comments.

File: session1/solutions/assignment4/Rakefile (hunk 7 above)
Comment (lirany1) [reply session1/solutions/assignment4/Rakefile]: I did it like in the Example :
Calling the command "rake archive PATH=/home/user/my_archive EXCLUDE=.tmp,.bkp" 
see the screen shot


do i need to change the code to the syntax of the input that you tried?

File: session1/solutions/assignment4/Rakefile (hunk 5 above)
Comment (lirany1) [reply session1/solutions/assignment4/Rakefile]: When i deleted the changing dir command, it compress the all folders in the path

File: session1/solutions/assignment4/Rakefile (hunk 8 above)
Comment (lirany1) [reply session1/solutions/assignment4/Rakefile]: Just for the user to see what  files he has in the folder
do i need to delete it?

File: session1/solutions/assignment4/Rakefile (hunk 4 above)
Comment (lirany1) [reply session1/solutions/assignment4/Rakefile]: Done :+1:

File: session1/solutions/assignment4/Rakefile (hunk 6 above)
Comment (lirany1) [reply session1/solutions/assignment4/Rakefile]: done,removed all the comments

File: session1/solutions/assignment3/Rakefile (hunk 1 above)
Comment (lirany1) [reply session1/solutions/assignment3/Rakefile]: I changed it to the correct one

File: session1/solutions/assignment3/Rakefile (hunk 2 above)
Comment (lirany1) [reply session1/solutions/assignment3/Rakefile]: "patt" is the user path to the file we want to change
i changed it to something more relevant

**Review by lirany1 — COMMENTED** [review]
Hi
@eyalstoler 
check please the changes according to your code review

**Pull request merged**